foca/api/register_openapi.py

```diff
@@ -0,0 +1,136 @@
+"""Functions for registering OpenAPI specs with a Connexion app instance."""
+
+from json import load, decoder
+import logging
+import os
+from typing import Dict, List, Optional
+
+from connexion import App
+from yaml import safe_load, safe_dump, parser
+
+from foca.config.config_parser import get_conf
+from foca.errors.errors import InvalidSpecification
+
+
+# Get logger instance
+logger = logging.getLogger(__name__)
+
+
+class OpenAPIConfig(Dict):
+    """Helper class for the configuration parameters of the specification files.
+    Inherits from the Dictionary class.
+
+    Attributes (optional):
+        out_file: The directory for the output specification file, in case it's

```

Comment:
This class is not easy to understand. Please add more extensive documentation for each of the option, as described in the corresponding issue. It must be clear how this can be used correctly, and in what context. Might be good to give a short example.